index.d.ts

```diff
@@ -0,0 +1,73 @@
+import EventEmitter from 'eventemitter3';
+
+interface IActionCable {
+  createConsumer: (url: string) => IConsumer;
+  startDebugging: () => void;
+  stopDebugging: () => void;
+  log: (messages: string[]) => void;
+}
+interface IConsumer {
+  subscriptions: ISubscriptions;
+  connection: IConnection;
+
+  send: (data: any) => void;
+  connect: () => void;
+  disconnect: () => void;
+  ensureActiveConnection: () => void;
+  createWebSocketURL: (url: string) => void;
+}
+interface ISubscriptions {
+  new (consumer: IConsumer): ISubscriptions;
+
+  consumer: IConsumer;
+  subscriptions: ISubscription[];
+
+  create: (channel: string | {channel: string}) => ISubscription;
```

Comment:
This is not correct as you can pass parameters to the channel, for example `create({ channel: "ChatChannel", room_id: 45 })`

See documentation here: https://guides.rubyonrails.org/action_cable_overview.html#subscriber

Reply:
Will this work? @renchap 
`create(channel: string | {channel:string} & Record<string, any>) => ISubscription;`